### PR TITLE
Issue #7604:Update NonEmptyAtclauseDescription

### PR DESCRIPTION
**Pull request by coderdeadpool**
Issue: #7604
Added example for each existing config to show what code is ok and what code cause violation.
```mvn clean verify``` Passed.

WebPage:

Screenshot (51)

First example:
Screenshot (52)

CLI o/p:
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="PARAM_LITERAL,DEPRECATED_LITERAL,THROWS_LITERAL,RETURN_LITERAL"/>
    </module>
  </module>
</module>
```
```
$ cat test.java
class Test
{
  // Violation for "param b", "deprecated", "throws" and "return" .
  /**
  * Some summary.
  * @param a Some description
  * @param b
  * @deprecated
  * @throws Exception
  * @return
  */
  public int method(String a, int b) throws Exception {
    return 1;
  }
}

```
```
$ java -jar target/checkstyle-10.9.2-SNAPSHOT-all.jar -c config.xml test.java
Starting audit...
[ERROR] D:\checkstyle\test.java:7: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] D:\checkstyle\test.java:8: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] D:\checkstyle\test.java:9: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] D:\checkstyle\test.java:10: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
Audit done.
Checkstyle ends with 4 errors.

```

Second example:

Screenshot (73)

CLI o/p:

```
$ cat config1.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="PARAM_LITERAL,THROWS_LITERAL"/>
    </module>
  </module>
</module>
```
```
$ cat test1.java
class Test
{
  // Violation for param "b" and at tag "throws".
  /**
  * Some summary.
  * @param a Some description
  * @param b
  * @deprecated
  * @throws Exception
  * @return
  */
  public int method(String a, int b) throws Exception {
    return 1;
  }
}
```
```
$ java -jar target/checkstyle-10.9.2-SNAPSHOT-all.jar -c config1.xml test1.java
Starting audit...
[ERROR] D:\checkstyle\test1.java:7: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] D:\checkstyle\test1.java:9: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
Audit done.
Checkstyle ends with 2 errors.
```

### CODE REVIEW

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani Added changes in this PR.
Please review the changes.

**Comment by romani** [comment]
>test.java:9: Javadoc comment at column 0 has parse error. Details: no viable alternative at input '  *' while parsing JAVADOC_TAG [NonEmptyAtclauseDescription]

There should non parse error. It should be violation message that point to empty tag.

>Some javadoc // OK

Remove `// OK`

**Comment by coderdeadpool** [comment]
@romani Remove it only from CLI o/p ?

**Comment by romani** [comment]
from your PR, all I care is user faced documentation. CLI is just prove that it works.

**Comment by romani** [comment]
ok we have a bug in parser.....
make your example code like this to bypass it (change in order and change in content - use my version):
```
rivanov@p5510:/var/tmp$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
       <module name="NonEmptyAtclauseDescription"/>
    </module>

</module>

/var/tmp$ cat Test.java 
class Test
{
  // Violation for "param b", "deprecated", "throws", "return" .
  /**
  * Some summary.
  * @param a Some description
  * @param b
  * @deprecated
  * @throws Exception
  * @return
  */
  public int method(String a, int b) throws Exception {
    return 1;
  }
}

/var/tmp$ java -jar checkstyle-10.9.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:7: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] /var/tmp/Test.java:8: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] /var/tmp/Test.java:9: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] /var/tmp/Test.java:10: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
Audit done.
Checkstyle ends with 4 errors.
```

**Comment by coderdeadpool** [comment]
@romani I got your point. But it was mentioned in #7604 description that we have to show what code is ok and what code cause violation.
That it why i did not removed it earlier.

**Comment by romani** [comment]
yes, you will use it in all cases, but `return` should be last tag in javadoc until #12864

**Comment by coderdeadpool** [comment]
@romani do i have to update the changes or i have to do it with CLI only ?

**Comment by coderdeadpool** [comment]
@romani Update the CLI o/p in the PR description. Please review it

**Comment by romani** [comment]
attentively read my sample: no "//OK", location of '{' ...

**Comment by coderdeadpool** [comment]
> attentively read my sample: no "//OK", location of '{' ...
> 
> but yes, direction is good, as thereis no parse errors

So shall i change it again with no "//ok" ?

**Comment by romani** [comment]
yes, lets point only on problematic lines

**Comment by coderdeadpool** [comment]
@romani updated PR with no "OK" and location of "{".

**Comment by coderdeadpool** [comment]
> yes, lets point only on problematic lines good description should "some good description"

Sorry i did get it !

**Comment by romani** [comment]
jsut do `* @param a Some description` as I pointed in my sample

**Comment by coderdeadpool** [comment]
On it.

**Comment by romani** [comment]
> On it.

dont do such comments, it is given, and makes noise, do comments only where work is done.

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani Please review the changes.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java [hunk 1]
```diff
@@ -67,44 +67,158 @@
...
+ *   return 1;
+ *  }
  * }
+ * </pre>
+ * <p>
+ * To configure the check to validate {@code @return} tags:
+ * </p>
+ * <pre>
+ * &lt;module name="NonEmptyAtclauseDescription"&gt;
+ *   &lt;property name="javadocTokens" value="RETURN_LITERAL"/&gt;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:117]: Make this config for param and expection tags. 

Remove all other example below this example. We do not need them. We just need to show users how to do customization on few tags , to show that coma is separator.

**Review by romani — CHANGES_REQUESTED** [review]
Last:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java (hunk 1 above)
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:117]: @romani We need to keep the examples only for the default check, and only for the @param and @exception tags, and remove all other examples. Is that what you are saying?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:117]: Yes

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani Added changes. Please review.

**Comment by romani** [comment]
misunderstanding again .....there should be 2 examples:
1. default
2. with `<property name="javadocTokens" value="PARAM_LITERAL,THROWS_LITERAL"/`

**Comment by coderdeadpool** [comment]
> misunderstanding again .....there should be 2 examples:
> 
> 1. default
> 2. with `<property name="javadocTokens" value="PARAM_LITERAL,THROWS_LITERAL"/`

@romani Sorry for misunderstanding, i thought we need seperate examples.
And are you sure ? because you said param and exception tags.
dont we have to use this ``` <property name="javadocTokens" value="PARAM_LITERAL,EXCEPTION_LITERAL"/ ``` ?.. the exception literal in place of throws literal ?

**Comment by romani** [comment]
>EXCEPTION_LITERAL

There is no such literal in javadoc

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
@romani Please review the changes.

**Review by romani — APPROVED** [review]
Ok to merge if CI is green

**Pull request merged**